src/lib/server/xlat_eval.c

```diff
@@ -284,25 +284,15 @@ static xlat_action_t xlat_eval_one_letter(TALLOC_CTX *ctx, fr_cursor_t *out, REQ
 	 */
 
 	case 'c': /* Current epoch time seconds */
-	{
-		struct timeval now;
-
-		gettimeofday(&now, NULL);
-
 		MEM(value = fr_value_box_alloc(ctx, FR_TYPE_UINT64, NULL, false));
-		value->datum.uint64 = (uint64_t)now.tv_sec;
-	}
+		value->datum.uint64 = (uint64_t)fr_time_to_sec(fr_time());
+
 		break;
 
 	case 'C': /* Current epoch time microsecond component */
-	{
-		struct timeval now;
-
-		gettimeofday(&now, NULL);
-
 		MEM(value = fr_value_box_alloc(ctx, FR_TYPE_UINT32, NULL, false));
```

Comment:
I know this one wasn't you, but the box should be a UINT64 ```MEM(value = fr_value_box_alloc(ctx, FR_TYPE_UINT64, NULL, false));```

Reply:
Also `s/value->datum.uint32/value->datum.uint64/g`